### PR TITLE
feat(spanner): [DO NOT MERGE]  add custom replication topology samples

### PR DESCRIPTION
**Pull request by rahul2393**
## Description
Added samples for supporting customer managed instance configurations

## Checklist
- [x] **Tests** pass
- [x] **Lint** pass: `bundle exec rubocop`
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: spanner/spanner_custom_instance_config.rb [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+def create_instance_config project_id:, instance_config_id:
+  # [START spanner_create_instance_config]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_config_id = "The customer managed instance configuration id. The id must start with 'custom-'"
+
+  require "google/cloud/spanner"
+  require "google/cloud/spanner/admin/instance"
+
+  instance_admin_client = Google::Cloud::Spanner::Admin::Instance::V1::InstanceAdmin::Client.new do |config|
+    # TODO: remove endpoint before GA
```
Comment (bajajneha27) [inline spanner/spanner_custom_instance_config.rb:24]: Are we planning to raise another PR before GA?

File: spanner/spanner_custom_instance_config.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    config.endpoint = "staging-wrenchworks.sandbox.googleapis.com"
+  end
+
+  instance_config_path = instance_admin_client.instance_config_path \
+    project: project_id, instance_config: instance_config_id
+  config = instance_admin_client.get_instance_config name: instance_config_path
+
+  labels_map = Google::Protobuf::Map.new :string, :string
```
Comment (bajajneha27) [inline spanner/spanner_custom_instance_config.rb:89]: Does this have to be a `Protobuf Map` or can it be a normal hash?

**Review by SandeepTuniki — CHANGES_REQUESTED** [review]

File: spanner/spanner_custom_instance_config.rb [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+  if job.error?
+    puts job.error
+  else
+    puts "Created instance configuration #{instance_config_id}"
+  end
+  # [END spanner_create_instance_config]
+end
+
+def update_instance_config project_id:, instance_config_id:
```
[the remark this answers is not in the document]
Comment (SandeepTuniki) [reply spanner/spanner_custom_instance_config.rb:72]: Something like #956 should do.

**Comment by bajajneha27** [comment]
Hi @rahul2393 
Do we still want to continue working on this?

**Comment by rahul2393** [comment]
@bajajneha27 you can close this PR, once the protos are available I will rebase and create a new PR and link this one for reference.

**Pull request closed without merging**